cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -1323,3 +1331,36 @@ fromAlonzoPParams
     , protocolParamMaxCollateralInputs = Just _maxCollateralInputs
     }
 
+-- | Calculate cost of making a UTxO entry for a given 'Value' and
+-- mininimum UTxO value derived from the 'ProtocolParameters'
+calcMinimumDeposit :: ProtocolParameters -> Value -> Lovelace
+calcMinimumDeposit pp v  =
+  case protocolParamMinUTxOValue pp of
+    Nothing ->
+      case protocolParamUTxOCostPerWord pp of
+        Nothing ->
+          error "calcMinimumDeposit: uTxOCostPerWord and minUTxOValue protocol parameters were not found"
+        Just (Lovelace cPerWord) ->
+          -- TODO: Alonzo - ledger should expose a function to calculate the required deposit
+          Lovelace $ max adaOnlyUTxOSize (utxoEntrySizeWithoutVal + Val.size (toMaryValue v)) * cPerWord
+    Just minUTxOValue ->
+      fromShelleyLovelace $ Shelley.scaledMinDeposit (toMaryValue v) (toShelleyLovelace minUTxOValue)
+
+ where
+    -- TODO: Copied from ledger. Ledger should expose a function for this
```

Comment:
TODO item for discussion

Reply:
this is a part of `utxoEntrySize` mentioned above